src/main/index.ts

```diff
@@ -474,6 +501,13 @@ ipcMain.on('post', (_: Event, notification: string, ...data: any[]): void => { /
     return;
   }
 
+  if (notification === 'saveLog') {
+    data.forEach((messages): void => {
+      saveLog(messages);
+    });
+    // saveLog(data);
```

Comment:
instead of this logic, lets just check if the first element of data is defined and is a string (which should be the `messages` parameter), and if so, just call `saveLog`

```ts
if (data.length > 0 && typeof data[0] === 'string') {
  saveLog(data[0]);
}
```